gtsam/discrete/DecisionTree-inl.h

```diff
@@ -625,13 +649,23 @@ namespace gtsam {
 
   template<typename L, typename Y>
   DecisionTree<L, Y> DecisionTree<L, Y>::apply(const Unary& op) const {
+    // It is unclear what should happen if tree is empty:
+    if (empty()) {
+      throw std::runtime_error(
+          "DecisionTree::apply(unary op) undefined for empty tree.");
+    }
```

Comment:
If the tree is empty, should this just be a no-op, which means we can return `this`?

Reply:
No. It is not allowed. *this would be wrong type.